### PR TITLE
SMART: Add support for NVMe SSD devices (Smart-Soft)

### PR DESCRIPTION
**Pull request by Sergey-Kirpa**
#1384 #1415 #1420 #1429
Unfortunately, I do not have any samsung 970 evo plus NVMe M.2 250G or any other similar device, and therefore I can’t check how it works. I hope it will work.

### CODE REVIEW

File: sysutils/smart/src/opnsense/service/conf/actions.d/actions_smart.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [list]
-command:/bin/ls /dev | grep '^\(ad\|da\|ada\)[0-9]\{1,2\}$'
+command:/usr/local/opnsense/scripts/OPNsense/Smart/list.sh
```
Comment (fabianfrz) [inline sysutils/smart/src/opnsense/service/conf/actions.d/actions_smart.conf]: wouldn't it make sense to just put it in the same call as the previous by just adding nvme via an or to the list?

**Comment by payback007** [comment]
the proposed code will not work I think. there are two missing/wrong commands. Please find attached my solution for line 43-46:

for dev in `ls /dev | grep '^nvme[0-9]\+$'`; do
    ident=`/usr/local/sbin/smartctl -a /dev/$dev | grep 'Serial Number' | awk -F ': *' '{print $2}'`;
    state=`/usr/local/sbin/smartctl -H /dev/$dev | grep 'SMART overall-health self-assessment test result' | awk -F ': *' '{print $2;exit}
/^SMART Health Status/ {print $2;exit}'`;

overall there was a missing **/dev/** and one space sign printed in the status, so passed was reported as a red sign in dashboard.

**Comment by Sergey-Kirpa** [comment]
> overall there was a missing **/dev/** and one space sign printed in the status, so passed was reported as a red sign in dashboard.

@payback007 
Thanks for your feedback. Could you attach the full output of the "/usr/local/sbin/smartctl -H /dev/$dev" command so that I can debug the script.

**Comment by fichtner** [comment]
could already work in the next version 2.1, as far as these changes go the PR is incompatible now.

**Pull request closed without merging**